From 57740a2200b107a7fe0954282ee32e28b10ddeb2 Mon Sep 17 00:00:00 2001
From: Pawel Zalewski <pzalewski@thegoodpenguin.co.uk>
Date: Fri, 7 Jul 2023 15:21:23 +0100
Subject: [PATCH 02/10] sysinfo: fix build with musl libc

If we use <sys/sysinfo.h> and a musl libc we will get
redefinition error of the sysinfo struct due to <linux/kernel.h>

Include <linux/const.h> and <linux/types.h> directly instead.

Signed-off-by: Pawel Zalewski <pzalewski@thegoodpenguin.co.uk>
---
 include/sysinc.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/sysinc.h b/include/sysinc.h
index 75ba1b59629..85237a3d950 100644
--- a/include/sysinc.h
+++ b/include/sysinc.h
@@ -131,7 +131,8 @@
 #endif
 
 #ifdef HAVE_LINUX_KERNEL_H
-#	include <linux/kernel.h>
+#	include <linux/const.h>
+#   include <linux/types.h>
 #endif
 
 #ifdef HAVE_ARPA_NAMESER_H
-- 
2.47.1

